pkgs/by-name/su/surrealist/package.nix

```
@@ -35,123 +34,91 @@ let
           owner = "evanw";
           repo = "esbuild";
           rev = "v${version}";
-          hash = "sha256-mED3h+mY+4H465m02ewFK/BgA1i/PQ+ksUNxBlgpUoI=";
+          hash = "sha256-h/Vqwax4B4nehRP9TaYbdixAZdb1hx373dNxNHvDrtY=";
         };
         vendorHash = "sha256-+BfxCyg0KkDQpHt/wycy/8CTG6YBA/VJvJFhhzUnSiQ=";
       });
   };
 
-  wasm-bindgen-cli-2-92 = wasm-bindgen-cli.override {
-    version = "0.2.92";
-    hash = "sha256-1VwY8vQy7soKEgbki4LD+v259751kKxSxmo/gqE6yV0=";
-    cargoHash = "sha256-aACJ+lYNEU8FFBs158G1/JG8sc6Rq080PeKCMnwdpH0=";
-  };
-
 in stdenv.mkDerivation (finalAttrs: {
   pname = "surrealist";
-  version = "1.11.7";
+  version = "2.0.3";
 
   src = fetchFromGitHub {
-    owner = "StarlaneStudios";
-    repo = "Surrealist";
-    rev = "v${finalAttrs.version}";
-    hash = "sha256-1jTvbr7jFo2GOB79ClwtBVVnNQlSEkqY2eqbiZxWG74=";
+    owner = "surrealdb";
+    repo = "surrealist";
+    rev = "surrealist-v${finalAttrs.version}";
+    hash = "sha256-+uOv/HyUitX4+XQc7SeueT0StEnI0v2Q9BgKyjH0m0k=";
   };
 
   sourceRoot = "${finalAttrs.src.name}/src-tauri";
 
-  embed = stdenv.mkDerivation {
-    inherit (finalAttrs) src version;
-    pname = "${finalAttrs.pname}-embed";
-    sourceRoot = "${finalAttrs.src.name}/src-embed";
-    auditable = false;
-    dontInstall = true;
-
-    cargoDeps = rustPlatform.fetchCargoTarball {
-      inherit (finalAttrs) src;
-      sourceRoot = "${finalAttrs.src.name}/src-embed";
-      hash = "sha256-0cAhaeoP8EPcE1230CyznQZZIKRs0lrI8XOXECgb8pg=";
-    };
-
-    nativeBuildInputs = [
-      cargo
-      rustc
-      llvmPackages_15.clangNoLibc
-      llvmPackages_15.lld
-      rustPlatform.cargoSetupHook
-      wasm-bindgen-cli-2-92
-    ];
-
-    postBuild = ''
-      CC=clang CARGO_TARGET_WASM32_UNKNOWN_UNKNOWN_LINKER=lld cargo build \
-        --target wasm32-unknown-unknown \
-        --release
-
-      wasm-bindgen \
-        target/wasm32-unknown-unknown/release/surrealist_embed.wasm \
-        --out-dir $out \
-        --out-name surrealist-embed \
-        --target web
-    '';
-  };
-
   pnpm-deps = stdenvNoCC.mkDerivation {
     inherit (finalAttrs) src version;
     pname = "${finalAttrs.pname}-pnpm-deps";
-    dontFixup = true;
 
     nativeBuildInputs = [ cacert jq moreutils nodePackages.pnpm ];
 
-    postInstall = ''
+    installPhase = ''
+      runHook preInstall
+
       export HOME=$(mktemp -d)
       pnpm config set store-dir $out
-      # use --ignore-script and --no-optional to avoid downloading binaries
-      # use --frozen-lockfile to avoid checking git deps
-      pnpm install --frozen-lockfile --no-optional --ignore-script
+      pnpm install --frozen-lockfile --ignore-script
```

```suggestion
      # Some packages produce platform dependent outputs. We do not want to cache those in the global store
      pnpm config set side-effects-cache false
      # --force allows us to fetch all dependencies including ones that aren't meant for our host platform
      pnpm install --frozen-lockfile --ignore-script --force
```
this should remove the need for architecture specific outputHashes. see #290715